File: samples/BOTH/async/Ping_Pong_2/pingpong.c
#include <unistd.h>

int perank;
int count = 0;

jactivity_t *pingserver(int);



jactivity_t *regme(char *, jcallback);

void regcallback(char *msg) {
  if (msg != NULL)
    perank = atoi(msg);
  else
    perank = -1;

  printf("Perank %d\n", perank);

  while(1) {
     usleep(1);
     printf("Pinging %d, count %d\n", perank, count++);
     pingserver(perank);
  }
}


int main() {
  printf("Registering...");
  regme("hello", regcallback);
  return 0;
}

File: samples/BOTH/async/Ping_Pong_2/pingpong.js
var count = 0;

jasync function pingserver(penum) {
    console.log("Ping received from ", penum);
}


jasync function regme(msg, cback) {
    
    count = count + 1;
    console.log("registration received from ", msg);
    cback('' + count);
}

